tests/controllers/test_rss_controller.py

```diff
@@ -141,10 +182,56 @@ def test_try_parse_feed_with_beautiful_soup(feed_input: str, is_url: bool, enabl
 
 @pytest.mark.parametrize("enable_cache", [True, False])
 def test_enable_disable_cache(enable_cache):
-    feed_input = "https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml"
-    rss_controller = RSSController(feed_input=feed_input, enable_cache=enable_cache)
+    controller = RSSController(feed_input=test_urls, enable_cache=enable_cache)
 
     if enable_cache:
-        assert rss_controller._session
+        assert controller._session
```

Comment:
Unit tests should not check internal class behavior; this breaks concepts of encapsulation and produces brittle tests, because it becomes possible to make changes to the class which are consistent and correct with respect to the API, but still fail.

If we want _session to be part of the public facing API, then it should be explicit. This could be as simple as something like adding a `cache_enabled()` method as well.

Reply:
Got it. I will remove private variable access from the tests.